### PR TITLE
Feature/condo/doma 1183/render extra info for employee selector in ticket form

### PR DESCRIPTION
**Pull request by AntonAL**
image

On create ticket page, employees in select list for "Responsible" and  "Executor" fields gets rendered with specialization.

For "Executor" field employees are splitted in following groups:
1. Technicians with matched specialization, belonging to matched division;
2. Technicians with matched specialization, belonging to other matched divisions;
3. Rest of employees.

Upgraded `GraphQlSearchInput` allows to render grouped options with `renderOptions` prop.

### CODE REVIEW

File: apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+            otherTechniciansOptions = differenceBy(employeeOptions, [
+                ...techniciansOnDivisionOptions,
+                ...techniciansOnOtherDivisionsOptions,
+            ], 'value')
+        }
+        const result = []
+        if (techniciansOnDivisionOptions.length > 0) {
+            result.push(
+                <Select.OptGroup label="Техники на участке">
```
Comment (dkoviazin) [inline apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx:96]: Move to translates

File: apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+                    {techniciansOnDivisionOptions.map(renderOption)}
+                </Select.OptGroup>
+            )
+        }
+        if (techniciansOnOtherDivisionsOptions.length > 0) {
+            result.push(
+                <Select.OptGroup label="Техники на других участках">
```
Comment (dkoviazin) [inline apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx:103]: Translates

File: apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+                    {techniciansOnOtherDivisionsOptions.map(renderOption)}
+                </Select.OptGroup>
+            )
+        }
+        if (otherTechniciansOptions.length > 0) {
+            result.push(
+                <Select.OptGroup label="Остальные сотрудники">
```
Comment (dkoviazin) [inline apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx:110]: to lang

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx (hunk 1 above)
Comment (AntonAL) [reply apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx]: Translated in `dd975bdc`.
Thank you for pointing on this.

File: apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx (hunk 2 above)
Comment (AntonAL) [reply apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx]: Translated in `dd975bdc`.

File: apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx (hunk 3 above)
Comment (AntonAL) [reply apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx]: Translated in `dd975bdc`.

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    )
+
+    const convertToOption = (employee) => ({
+        text: employee.name,
+        value: employee.user.id,
+        data: employee,
+    })
+
+    /**
+     * Employees are grouped by following rules:
```
Comment (Dimitreee) [inline apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx:60]: Should it be client side only logic? Maybe we should move this to serverce?

**Review by Dimitreee — APPROVED** [review]

File: apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx (hunk 4 above)
Comment (AntonAL) [reply apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx:60]: This use case is only presented in CRM desktop app for Organisations yet in ticket form and this is a semi-automatic feature, that just pre-selects executor and responsible, allowing user to change selection while creating a ticket.

Comment (Dimitreee) [reply apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx:60]: All feature with divisions is introduced only for ability to preselect assignee and executor;) I'm not sure this is client side only logic.
Guess we need to introduce dedicated service for this.

Comment (AntonAL) [reply apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx:60]: I agree, I will extract grouped suggestions of executors into custom GraphQL query.
Created separate task DOMA-1212 for it.
Thank You for this point.

**Pull request merged**